### PR TITLE
[leaflet] Update to leaflet 1.9.1

### PR DESCRIPTION
**Pull request by someonewithpc**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by someonewithpc** [comment]
## Breaking Changes
Expose ESM entrypoint with Leaflet global (https://github.com/Leaflet/Leaflet/pull/8329 by @jonkoops). If you're using Leaflet in a ESM setup with plugins that rely on the L global, you have to use the default import for them to work:
```diff
-import * as L from 'leaflet';
+import L from 'leaflet';
```
 - [ ] Update `color-adjust` to `print-color-adjust` (https://github.com/Leaflet/Leaflet/pull/8211 by @Malvoz)

## API changes
 - [X] Add content and latLng options to Popup / Tooltip constructors (https://github.com/Leaflet/Leaflet/pull/7783 by @Falke-Design)
 - [ ] Extend Bounds to have the same functions as LatLngBounds (https://github.com/Leaflet/Leaflet/pull/7882 by @Falke-Design)

## Improvements
 - [ ] Update getCenter() calculation and move it to PolyUtil / LineUtil (https://github.com/Leaflet/Leaflet/pull/7603 by @Falke-Design)
 - [ ] Remove border styles in overflowing popups (https://github.com/Leaflet/Leaflet/pull/8260 by @Malvoz)
 - [ ] Fix "listener not found" warning when setting maxBounds (https://github.com/Leaflet/Leaflet/pull/8168 by @mourner)
 - [ ] Remove "listener not found" warning (https://github.com/Leaflet/Leaflet/pull/8234 by @Falke-Design)
 - [ ] Extend Events.listens to search for specific function (https://github.com/Leaflet/Leaflet/pull/8161 by @Falke-Design)
 - [ ] Add noMoveStart option to panTo (https://github.com/Leaflet/Leaflet/pull/6685 by @Chivano)
 - [ ] Add FeatureCollection handling to geometryToLayer (https://github.com/Leaflet/Leaflet/pull/8163 by @Falke-Design)
raised_hands Accessibility
 - [ ] Improve Tooltip accessibility (focus and voice over) (https://github.com/Leaflet/Leaflet/pull/8247 by @alekzvik)
 - [ ] Fix links in accessibility guide (https://github.com/Leaflet/Leaflet/pull/8198 by @Malvoz)
 - [ ] Remove role="presentation" from image tiles (https://github.com/Leaflet/Leaflet/pull/8172 by @Malvoz)
lady_beetle Bug fixes
 - [ ] Fix invalid GeoJSON on unbalanced arrays (https://github.com/Leaflet/Leaflet/pull/7637 by @steff1986)
 - [ ] Fix 2 step zooming while using mouse wheel scrolling (https://github.com/Leaflet/Leaflet/pull/8298 by @Falke-Design)
 - [ ] Fix wrong assigned parameter while calling map._move over requestAnimFrame (https://github.com/Leaflet/Leaflet/pull/8328 by @AMDvsTMD)
 - [ ] Fix _isClickDisabled to not throw no error if parent is removed from DOM (https://github.com/Leaflet/Leaflet/pull/8288 by @Falke-Design)
 - [ ] Fix DomEvent.DoubleTap to ignore clicks on <label>s with a for attribute (https://github.com/Leaflet/Leaflet/pull/8227 by @IvanSanchez)
 - [ ] Fix calling once() twice if same event is fired inside once (https://github.com/Leaflet/Leaflet/pull/8190 by @Falke-Design)
 - [ ] Fix map.getCenter() returning a mutable object (https://github.com/Leaflet/Leaflet/pull/8167 by @mourner)
 - [ ] Fix regression about popup close button modifying the URL (https://github.com/Leaflet/Leaflet/pull/8160 by @IvanSanchez)
 - [ ] Fix min/maxZoom when used in combination with detectRetina (https://github.com/Leaflet/Leaflet/pull/7328 by @bozdoz)
memo Docs
 - [ ] Use preferred tile.openstreetmap.org URL (https://github.com/Leaflet/Leaflet/pull/8418 by @Firefishy)
 - [ ] Use LocalStorage for dialog sessions (https://github.com/Leaflet/Leaflet/pull/8382 by @ChristopherWirtOfficial)
 - [ ] Update anchor links for headers and in collapsed accordions (https://github.com/Leaflet/Leaflet/pull/7780 by @Falke-Design)
 - [ ] Fix typo in reference-1.6.0.html (https://github.com/Leaflet/Leaflet/pull/8330 by @eltociear)
 - [ ] Add pre-commit linting to CONTRIBUTING.md (https://github.com/Leaflet/Leaflet/pull/8299 by @Falke-Design)
 - [ ] Ensure no borders on dialog iframe (https://github.com/Leaflet/Leaflet/pull/8296 by @Malvoz)
 - [ ] Replace Mapbox with OpenStreetMap in tutorials and examples (https://github.com/Leaflet/Leaflet/pull/7818 by @Falke-Design)
 - [ ] Remove DOCS-TODO.md (https://github.com/Leaflet/Leaflet/pull/8259 by @Malvoz)
 - [ ] Better PosAnimation example (https://github.com/Leaflet/Leaflet/pull/7386 by @stell)
 - [ ] Correct heading level in GeoJSON example (https://github.com/Leaflet/Leaflet/pull/8230 by @Malvoz)
 - [ ] Update Overlay Tutorial (ImageOverlay, VideoOverlay, SVGOverlay) (https://github.com/Leaflet/Leaflet/pull/8090 by @KonstantinBiryukov)
 - [ ] Change attribute anchor to data-anchor (https://github.com/Leaflet/Leaflet/pull/8174 by @KnightJam1)
 - [ ] Fix bad markdown causing link to not work (https://github.com/Leaflet/Leaflet/pull/8156 by @freyfogle)
 - [ ] A couple of site SEO fixes (https://github.com/Leaflet/Leaflet/pull/8229 by @Malvoz)
 - [ ] Fix attribution flag 1px misalignment on some websites (https://github.com/Leaflet/Leaflet/issues/8170 by @mourner)
 - [ ] Attribution flag now resizes with font-size changes (https://github.com/Leaflet/Leaflet/pull/8183 by @sumitsaurabh927)
 - [ ] Add Dialog to website (https://github.com/Leaflet/Leaflet/pull/8177 by @Falke-Design and https://github.com/Leaflet/Leaflet/pull/8193, https://github.com/Leaflet/Leaflet/pull/8194 by @Malvoz)
wrench Workflow
 - [ ] Improve GitHub Workflows security (https://github.com/Leaflet/Leaflet/pull/8419 by @sashashura)
 - [ ] Update development dependencies
 - [ ] Replace deprecated eslint-plugin-script-tags (https://github.com/Leaflet/Leaflet/pull/8331 by @jonkoops)
 - [ ] Use major version ranges for Github Actions (https://github.com/Leaflet/Leaflet/pull/8286 by @jonkoops)
 - [ ] Configure YAML issue forms (https://github.com/Leaflet/Leaflet/pull/8246 by @Malvoz)
 - [ ] Add FUNDING.yml (@mourner)
 - [ ] Add pre-commit hook to fix linting issues (https://github.com/Leaflet/Leaflet/pull/8212 by @jonkoops)
 - [ ] Remove Dependabot specific labels (https://github.com/Leaflet/Leaflet/pull/8199 by @jonkoops)
 - [ ] Use shorter bundlemon names (https://github.com/Leaflet/Leaflet/pull/8195 by @mourner)
 - [ ] Make sure integrity hashes are generated for the built version (@mourner)
test_tube Tests
 - [ ] Added test cases for map.latLngToLayerPoint method (https://github.com/Leaflet/Leaflet/pull/8407 by @kreloaded)
 - [ ] Add test for map.panTo (https://github.com/Leaflet/Leaflet/pull/8390 by @anurag-dhamala)
 - [ ] Add test for map.containerPointToLatLng and map.latLngToContainerPoint (https://github.com/Leaflet/Leaflet/pull/8384 by @abhi3315)
 - [ ] Add test for Layer._addZoomLimit (https://github.com/Leaflet/Leaflet/pull/8037 by @zishiwu123)
 - [ ] Add tests for Map (https://github.com/Leaflet/Leaflet/pull/8206 by @stephenspol)
 - [ ] Add test for CircleMarker._containsPoint (https://github.com/Leaflet/Leaflet/pull/8340 by @gernhard1337)
 - [ ] Add missing handler tests (https://github.com/Leaflet/Leaflet/pull/8182 by @Falke-Design)
 - [ ] Cover Rectangle with unit Tests (https://github.com/Leaflet/Leaflet/pull/8144 by @stephenspol)

**Comment by Falke-Design** [comment]
> Expose ESM entrypoint with Leaflet global (https://github.com/Leaflet/Leaflet/pull/8329).

Will be reverted soon with 1.9.2

**Comment by someonewithpc** [comment]
> > Expose ESM entrypoint with Leaflet global ([Leaflet/Leaflet#8329](https://github.com/Leaflet/Leaflet/pull/8329)).
> 
> Will be reverted soon with 1.9.2

Didn't know, but I don't think this would require any change on our side, if I understand it correctly

**Review by Falke-Design — CHANGES_REQUESTED** [review]

**Comment by someonewithpc** [comment]
All tests pass but produce the following warning:

```
{"leaflet": [{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'PosAnimation', which is missing from the declaration module."},{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'geoJson', which is missing from the declaration module."},{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'noConflict', which is missing from the declaration module."}]}
```

But those properties are in the definition. Is this because the definition doesn't define a global `L`, or something else?

File: types/leaflet/index.d.ts [hunk 1]
```diff
@@ -56,10 +56,12 @@ export namespace LineUtil {
         useLastCode?: boolean,
         round?: boolean,
     ): [Point, Point] | false;
+    function polylineCenter(latlngs: LatLngExpression[], crs: CRS, zoom: number): LatLng;
```
[the remark this answers is not in the document]
Comment (someonewithpc) [reply types/leaflet/index.d.ts:59]: Wasn't mentioned in the changelog, but yep, you're right. Removing it

**Comment by Falke-Design** [comment]
> All tests pass but produce the following warning:
> 
> ```
> {"leaflet": [{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'PosAnimation', which is missing from the declaration module."},{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'geoJson', which is missing from the declaration module."},{"fileName":"/home/hugo/projects/DefinitelyTyped/types/leaflet/index.d.ts","ruleName":"npm-naming","message":"The declaration doesn't match the JavaScript module 'leaflet'. Reason:\nThe JavaScript module exports a property named 'noConflict', which is missing from the declaration module."}]}
> ```
> 
> But those properties are in the definition. Is this because the definition doesn't define a global `L`, or something else?

That is the reason why we will release 1.9.2 soon with out the new ESM entrypoint -> Breaks imports

**Comment by Falke-Design** [comment]
We also changed the API of `Events.listens` https://github.com/Leaflet/Leaflet/pull/8161

**Comment by someonewithpc** [comment]
> We also changed the API of `Events.listens` [Leaflet/Leaflet#8161](https://github.com/Leaflet/Leaflet/pull/8161)

@Falke-Design I saw that PR, but it didn't seem to require any changes to the declaration. Do please correct me if I'm wrong, but `listens` is either `(type: string, propagate?: boolean) => boolean` or `(type: string, fn: LeafletEventHandlerFn, context?: any, propagate?: boolean) => boolean` (see https://github.com/someonewithpc/DefinitelyTyped/blob/master/types/leaflet/index.d.ts#L496-L502 and https://github.com/someonewithpc/DefinitelyTyped/blob/master/types/leaflet/index.d.ts#L536)

**Review by Falke-Design — APPROVED** [review]
Yes sorry my fault. I didn't checked the file only your changes

**Review by jonkoops — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Comment by someonewithpc** [comment]
Ready to merge

**Pull request merged**